### ISSUE TITLE
implementation on efficientnet

### ISSUE THREAD

**Issue by ooodragon94** [he-y/filter-pruning-geometric-median#63]
Hello!
I've read your paper and implemented efficientnet with it! (I hope I didn't make any mistake on it; I've seen all kernel weights after pruning and it seems to work fine)

I used pretrained model [efficientnet-b0](https://github.com/lukemelas/EfficientNet-PyTorch) which gives 76.122 % on imagenet
My settings were 1 on norm pruning (no pruning done on norm) and 0.1 on dist pruning (10% pruning on dist method)
the performance degrades severely, giving only 33% and it seemed not to increase after some finetuning

I've just got result on pruning only 1% which gave me 74% accuracy. (oops, I'm trying to finetune it now and it is giving me very low accuracy: about 25%. Maybe I have lr too high but I will have to see)
I'm thinking (correct me if I am wrong) that when I prune 10% the zeroed filters are affecting later layers and even though later layers are not pruned, they have hard time learning due to result on pruned previous layers (and btw they are groupconv); may be similar to this [issue](https://github.com/he-y/filter-pruning-geometric-median/issues/7) 

I just wanted to ask you if this problem will alleviate  if I consecutively prune 1% and finetune 1 epoch until I reach 10 % of pruning ratio. Or if you have any other method I will be more than welcome to hear! thank you for reading!

additional note) I'm trying your code with pretrained Resnet, it first gave 56.9% acc and after one epoch of finetuning it gave 74.606%, and finetuning for another epoch did not change the result... still in 74.606%

**Comment by he-y** [comment]
Thanks for your interest.
I am not very familiar with efficientnet right now, but I guess the reason might be that the channel numbers of efficientnet are already optimized. Therefore,  pruning some filters means the channel numbers changes to some worse ones.
groupconv might be another reason. GM-based criterion and previous norm-based criterion are designed for normal conv. Some adjustments may be needed for groupconv.

**Comment by ooodragon94** [comment]
thanks for your good intuition!
hmm.. guess I need a break through with it..
